Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+   *
+   * @throws \RuntimeException if $table and $update properties aren't set.
+   */
+  private function buildUpdate() {
+    // `UPDATE` part.
+    if (!isset($this->table))
+      throw new RuntimeException('Cannot execute the query without a target table');
+
+    $query = 'UPDATE '.$this->table.' ';
+
+    // Append `SET` part.
+    if (!isset($this->update) || empty($this->update))
+      throw new RuntimeException('Cannot execute an UPDATE query without updating anything');
+
+    $query .= 'SET ';
+    $count = 0;
+    foreach ($this->update as $field => $value) {
+      if ($count != 0)
+        $query .= ', ';
+      $query .= $field.' = '.$value;
+      $count++;
+    }
+
+    // Append `WHERE` part.
+    $query .= ' '.$this->where;
+
+    // Append `ORDER BY` part.
+    $query .= $this->buildOrderBy();
+
+    // Append `LIMIT` part.
+    $query .= $this->buildLimit();
+
+    return $query;
+  }
+
+  /**
+   * Build INSERT query.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @return string
+   *
+   * @throws \RuntimeException if $table and $insert properties aren't set.
+   */
+  private function buildInsert() {
+    // `INSERT INTO` part.
+    if (!isset($this->table))
+      throw new RuntimeException('Cannot execute the query without a target table');
+    if (!isset($this->insert) || empty($this->insert))
+      throw new RuntimeException('Cannot execute an INSERT query without inserting anything');
+
+    $query = 'INSERT INTO '.$this->table.' (';
+
+    // Append fields and values.
+    $values = '';
+    $count = 0;
+    foreach ($this->insert as $field => $value) {
+      if ($count != 0) {
+        $query .= ', ';
+        $values .= ', ';
+      }
+      $query .= $field;
+      $values .= $value;
+      $count++;
+    }
+    $query .= ') VALUES ('.$values.')';
+
+    return $query;
+  }
+
+  /**
+   * Build DELETE query string.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @return string
+   *
+   * @throws \RuntimeException if $table property isn't set.
+   */
+  private function buildDelete() {
+    // `DELETE` part.
+    if (!isset($this->table))
+      throw new RuntimeException('Cannot execute the query without a target table');
+
+    $query = 'DELETE '.$this->delete_modifier.' FROM '.$this->table.' ';
+
+    // Append `WHERE` part.
+    $query .= $this->where;
+
+    // Append `ORDER BY` part.
+    $query .= $this->buildOrderBy();
+
+    // Append `LIMIT` part.
+    $query .= $this->buildLimit();
+
+    return $query;
+  }
+
+  /**
+   * Build ORDER BY query part.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @return string
+   */
+  private function buildOrderBy() {
+    $part = '';
+
+    if (!empty($this->order_by)) {
+      $part .= ' ORDER BY ';
+      $count = 0;
+      foreach ($this->order_by as $field => $direction) {
+        if ($count != 0)
+          $part .= ', ';
+        $part .= $field.$direction;
+      }
+    }
+
+    return $part;
+  }
```

Comment:
```php
private function buildOrderBy() {
  if (empty($this->order_by))
    return ''; // This can also be `return;` since `null` casts to empty string.
  
  return ' ORDER BY '.join(',', array_map(function($field, $direction) {
    return $field.$direction; // Assumes $direction has leading space.
  }, array_keys($this->order_by), array_values($this->order_by)));
}
```